k8s_test_harness/util/env_util.py

```diff
@@ -0,0 +1,204 @@
+            path: rockPath,
+            arch: arch,
+            image: image,
+            "rockcraft-revision": rockcraftRevisions[arch] || '',
+            "runs-on-labels": platformLabels[arch] || [inputs["runs-on"]]
+        }
+
+        :raises json.decoder.JSONDecodeError: on invalid JSON input.
+        :raises ValueError: on missing or null-valued fields.
+        """
+        return cls.from_dict(json.loads(json_str))
+
+    @property
+    def rockfile_path(self) -> str:
+        """Returns the path to the rockfile.yaml file for this image
+        based on the `path` property (RockMetaInfo.rock_dir) of the
+        ROCK's meta info.
+
+        Note that this can be a relative path with respect to the root
+        of the repository the `build_rocks.yaml` workflow had run on.
+        """
+        return os.path.join(self.rock_dir, "rockfile.yaml")
+
+
+def get_rocks_meta_info_from_env(
+        rock_metadata_env_var: str=DEFAULT_BUILT_ROCKS_METADATA_ENV_VAR
+) -> List[RockMetaInfo]:
+    """ Attempts to parse the `RockMetaInfo`s from the JSON string
+    from the env var with the given name as produced by the
+    canonical/k8s-workflows/build_rocks.yaml workflow:
+
+    # TODO(aznashwan): pin exact link upon first stable release of workflows:
+    https://github.com/canonical/k8s-workflows/blob/main/.github/workflows/build_rocks.yaml#L125-L133
+
+    :raises EnvironmentError: when the env variable isn't set or is empty.
+    :raises json.decoder.JSONDecodeError: on invalid JSON input.
+    :raises ValueError: on missing, null-valued, or malformed values.
+    """
+    rock_metas = must_get_env_var(rock_metadata_env_var)
+
+    rock_metas_list = json.loads(rock_metas)
+
+    return [RockMetaInfo.from_dict(d) for d in rock_metas_list]
+
+
+def get_builds_meta_info_for_rock(
+        rock_name: str,
+        rock_metadata_env_var: str=DEFAULT_BUILT_ROCKS_METADATA_ENV_VAR
+) -> List[RockMetaInfo]:
+    """Returns a list of all build meta info sets for the ROCK with
+    the given name (should be one entry per version and architecture
+    the ROCK was built for) from the ;environment variable produced
+    by the canonical/k8s-workflows/build_rocks.yaml workflow:
+
+    # TODO(aznashwan): pin exact link upon first stable release of workflows:
+    https://github.com/canonical/k8s-workflows/blob/main/.github/workflows/build_rocks.yaml#L125-L133
+
+    :raises EnvironmentError: when the env variable isn't set or is empty.
+    :raises json.decoder.JSONDecodeError: on invalid JSON input.
+    :raises ValueError: on missing, null-valued, or malformed values.
+    """
+    all_metas = get_rocks_meta_info_from_env(rock_metadata_env_var)
+
+    return [r for r in all_metas if r.name == rock_name]
+
+
+def get_build_meta_info_for_rock_version(
+        rock_name: str,
+        rock_version: str,
+        rock_arch: str,
+        rock_metadata_env_var: str=DEFAULT_BUILT_ROCKS_METADATA_ENV_VAR
+) -> RockMetaInfo:
```

Comment:
just a comment: this basically rebuilds the whole `RockMetaInfo` list from the environment every time you call this function. Maybe returning a `RockMetaInfos` data object would make sense and this object has a `get_info_for_rock_version`. 

Anyway, this is not performance critical code - just leave it like this.